### DISCUSSION TITLE
Cypress test: Infinitely appended iframe and HTTP error 414 - Request URL too long when Microsoft login, and page load timeout

### DISCUSSION THREAD

**Discussion by dphoenix** [cypress-io/cypress#25679] [Questions and Help]
Hello,

I am testing login for a site.  The login button redirects to a Microsoft login page when clicked.

Originally I had a frame busting problem, where the test would redirect outside of the Cypress framework, but solved that with the `experimentalSessionAndOrigin` attribute.

However, now I am facing two new errors.

First, I get an HTTP 414 error, with the URL being too long when Cypress clicks on the button.  I noticed that when I looked at the URL in the failure message, it seems that the url seems to have an `iframe-request-id` query parameter infinitely appended to the URL string, so I see why the URL is too long.  I just don't know why that string is appended like that.  

I should also note that the URL in the code is generated as an anchor tag, with the href element being a Vue a property.  

Also, and I noticed this in a few other tests as well when clicking to a new page, I get a page load timeout error.  I can understand that for the infinitely long URL, but my issue is that I set the pageLoadTimeout configuration setting to something insanely big (like 9e+67ms, just to make sure there's enough time).  And the timeout error happens after only a second or so.  I get this error with another page as well, which is a Vue route, and which does eventually load.

Also, the web portal I am testing is on my local computer, it's a Vue application that is dockerized.

Any ideas about where I start looking?  I can't really share a lot of the actual code because it's not mine and kind of private, but I can try to maybe mask some of it.

Thanks!

**Comment by jahunsbe-sj** [comment]
Try setting `experimentalModifyObstructiveThirdPartyCode` as described here: https://docs.cypress.io/guides/end-to-end-testing/azure-active-directory-authentication

**Comment by semics-tech** [comment]
I'm also having this issue, I've tried setting `experimentalModifyObstructiveThirdPartyCode` but still getting an infinite redirect with a lot of `iframe-request-id` query parameters.

I setup a new project and tried visiting the [Microsoft Login](https://login.microsoftonline.com/common/oauth2/v2.0/authorize?client_id=4765445b-32c6-49b0-83e6-1d93765276ca&redirect_uri=https%3A%2F%2Fwww.office.com%2Flandingv2&response_type=code%20id_token&scope=openid%20profile%20https%3A%2F%2Fwww.office.com%2Fv2%2FOfficeHome.All&sso_reload=true/) page. This is where the error seems to be.

Also, there is a discrepency in the documentation about where to put the `experimentalModifyObstructiveThirdPartyCode` parameter, which one is correct?

https://docs.cypress.io/guides/end-to-end-testing/azure-active-directory-authentication#Configuring-Cypress-to-use-Microsoft-AAD
```
export default defineConfig({
  e2e: {
    experimentalModifyObstructiveThirdPartyCode: true,
  },
})
```

https://docs.cypress.io/guides/guides/web-security#Modifying-Obstructive-Third-Party-Code

```
export default defineConfig({
  experimentalModifyObstructiveThirdPartyCode: true,
})
```

**Reply by frnc07** [reply]
The following worked for me:

Put `experimentalModifyObstructiveThirdPartyCode` outside of e2e.
Use cy.origin() as shown below

cy.origin('your secondary domain your app will redirect to', () => {
      cy.visit('your base url')
})

**Comment by souravs17031999** [comment]
@semics-tech @dphoenix @jahunsbe-sj did you guys find solution to it ? I am facing same issues...

**Comment by k-cypress** [comment]
I had the same problem but was able to solve it like this: 
https://stackoverflow.com/questions/70379955/unable-to-load-my-azure-hosted-web-app-through-cypress

**Reply by HanhNguyen102** [reply]
I have the same problem. Would you mind sharing your code?

**Comment by manu-remsense** [comment]
I have the same issue, but using SAML and not Oauth2.. if anyone has the solution, let me know :/

**Comment by lucasltv** [comment]
Same error here

**Comment by asagarelse** [comment]
Can anyone please help in resolving this issue?
Thanks

**Comment by Dileepa21** [comment]
Yes ,Can some one give a solid answer for this? with an example
